rust/gitxetcore/src/git_integration/repo_migration/migration.rs

```diff
@@ -0,0 +1,996 @@
+            if reference.is_note() && reference_name.starts_with("refs/notes/xet/") {
+                mg_trace!("Xet Note, rejecting.");
+                continue;
+            }
+
+            // Only convert local references
+            if reference.is_remote() {
+                mg_trace!("Xet Note, rejecting.");
+                continue;
+            }
+
+            if let Some(oid) = reference.target() {
+                mg_trace!("Importing reference {reference_name} with oid {oid}");
+                seed_oids.insert(oid);
+            }
+        }
+
+        // If it's not converting notes, also add in the tags.
+        if !converting_notes {
+            src.tag_foreach(|oid, _| seed_oids.insert(oid))?;
+        }
+
+        //////////////////////////////////////////////////////////////////////////////////////////
+        //
+        //  Step 1: build the dependency graph.  We track what upstream objects must be converted before
+        //  the object itself must be converted.
+        //
+        //  In addition, track a list of root oids that are referenced by commits
+        //  (except for note commits).
+
+        // After conversion, these dependent oids are closer to being ready.
+        let mut downstream_oids = HashMap::<Oid, Vec<Oid>>::new();
+
+        // The number of dependencies that have to be ready before we can convert this one.
+        let mut dependency_count = HashMap::<Oid, usize>::new();
+        let mut blobs = HashSet::new();
+        let mut root_tree_oids = HashSet::new();
+
+        // Start us off with the seed oids from above.
+        let mut proc_queue = Vec::from_iter(seed_oids.into_iter());
+
+        while let Some(oid) = proc_queue.pop() {
+            if dependency_count.contains_key(&oid) {
+                continue;
+            }
+            // Make sure this entry has a spot, even if there are no dependents.
+            downstream_oids.entry(oid).or_default();
+
+            progress_reporting.update_target(Some(1), None);
+
+            let Ok(obj) = src.find_object(oid, None).map_err(|e| {
+                mg_warn!(
+                    "Referenced Oid {oid} not found in src database, passing Oid through as is."
+                );
+                e
+            }) else {
+                dependency_count.insert(oid, 0);
+                continue;
+            };
+
+            let dependents = match obj.kind().unwrap_or(ObjectType::Any) {
+                ObjectType::Tree => {
+                    if converting_notes {
+                        get_note_tree_dependents(&src, obj, &full_tr_map)
+                    } else {
+                        get_nonnote_tree_dependents(obj)
+                    }
+                }
+                ObjectType::Commit => {
+                    let (dependents, tree_oid) = get_commit_dependents(&src, obj);
+                    if !converting_notes {
+                        // Commits reference the root trees.
```

Comment:
Add note -- these need to be handled separately so we know to replace or add the .gitattributes file.